Spatial_Engine/Query/IsEqualProperties.cs

```diff
@@ -0,0 +1,97 @@
+        /****        Public Methods            ****/
+        /******************************************/
+
+        [Description("Evaluates if the two elements non-geometrical data is equal to the point that they could be merged into one object.")]
+        [Input("element", "An IElement0D to compare the properties of with an other IElement0D.")]
+        [Input("other", "The IElement0D to compare with the other IElement0D.")]
+        [Output("equal", "True if the Objects non-geometrical property is equal to the point that they could be merged into one object.")]
+        public static bool IIsEqualProperties(this IElement0D element, IElement0D other)
+        {
+            return IIIsEqualProperties(element, other);
+        }
+
+        /******************************************/
+
+        [Description("Evaluates if the two elements non-geometrical data is equal to the point that they could be merged into one object.")]
+        [Input("element", "An IElement1D to compare the properties of with an other IElement1D.")]
+        [Input("other", "The IElement1D to compare with the other IElement1D.")]
+        [Output("equal", "True if the Objects non-geometrical property is equal to the point that they could be merged into one object.")]
+        public static bool IIsEqualProperties(this IElement1D element, IElement1D other)
+        {
+            return IIIsEqualProperties(element, other);
+        }
+
+        /******************************************/
+
+        [Description("Evaluates if the two elements non-geometrical data is equal to the point that they could be merged into one object.")]
+        [Input("element", "An IElement2D to compare the properties of with an other IElement2D.")]
+        [Input("other", "The IElement2D to compare with the other IElement2D.")]
+        [Output("equal", "True if the Objects non-geometrical property is equal to the point that they could be merged into one object.")]
+        public static bool IIsEqualProperties(this IElement2D element, IElement2D other)
+        {
+            return IIIsEqualProperties(element, other);
+        }
+
+        /******************************************/
+        /****        Private Methods           ****/
+        /******************************************/
+
+        private static bool IIIsEqualProperties(this IElement element, IElement other)
+        {
+            // Geometrical objects don't have properties
+            if (typeof(IGeometry).IsAssignableFrom(element.GetType()) && typeof(IGeometry).IsAssignableFrom(other.GetType()))
+                return true;
+
+            // Different types are different
+            if (element.GetType() != other.GetType())
+                return false;
+
+            // look for a specific comparing method
+            object result = Reflection.Compute.RunExtensionMethod(element, "IsEqualProperties", new object[] { other });
+
+            if (result == null)
```

Comment:
can be good to check if the result is of type bool here, to avoid potential crashes on the return line (should probably never ever happen, but still good to get it in).